### PR TITLE
Fix race condition of deleting ccnp in e2e test

### LINKED ISSUE #24380: Fix race condition of CRD deletion in the e2e test suits
As pointed out in [this comment](https://github.com/cilium/cilium/pull/23935#issuecomment-1467897146), e2e test for `K8sDatapathConfig Host firewall With VXLAN` is likely to hit a flake which is caused by race condition in the `AfterEach` where we delete the policy using `kubectl delete --ccnp` without waiting.

Something similar to https://github.com/cilium/cilium/pull/24097 can fix it.

I shall also check the e2e test thoroughly to eliminate the issues alike.

### PR DESCRIPTION
**Pull request by jschwinger233**
There is a flake in e2e test when a test case starts to proceed before ccnp/cnp comes to take effect by cilium-agent. The correct way to delete ccnp/cnp is to run "kubectl delete" followed by "cilium policy wait", and kubectl helper already has such wrappers.

Fixes: #24380

```release-note
Fix race conditions when deleting CNP / CCNP in e2e tests
```

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
Nice find!

**Comment by jschwinger233** [comment]
I believe the CI failure belongs to a known flake https://github.com/cilium/cilium/issues/20723, and I can't reproduce it by running locally.

**Comment by pchaigno** [comment]
> I believe the CI failure belongs to a known flake #20723, and I can't reproduce it by running locally.

I agree. Feel free to mark ready-to-merge once it's reviewed and you have triaged CI failures with a comment.

**Comment by julianwiedmann** [comment]
Let's keep :eyes: on the HostFW flakes (eg. https://github.com/cilium/cilium/issues/22578) for a bit ... maybe this is all what was needed for them?

@jschwinger233 we typically backport CI fixes. Could you check how far back this applies (I would assume v1.11), and add backport labels? :pray:

File: test/k8s/datapath_configuration.go [hunk 1]
```diff
@@ -649,18 +649,24 @@ func fetchPodsWithOffset(kubectl *helpers.Kubectl, namespace, name, filter, host
...
+func applyL3Policy(kubectl *helpers.Kubectl, ns string) (withdrawPolicy func()) {
 	demoPolicyL3 := helpers.ManifestGet(kubectl.BasePath(), "l3-policy-demo.yaml")
 	By(fmt.Sprintf("Applying policy %s", demoPolicyL3))
 	_, err := kubectl.CiliumPolicyAction(ns, demoPolicyL3, helpers.KubectlApply, helpers.HelperTimeout)
 	ExpectWithOffset(1, err).Should(BeNil(), fmt.Sprintf("Error creating resource %s: %s", demoPolicyL3, err))
+
+	return func() {
+		_, err := kubectl.CiliumPolicyAction(ns, demoPolicyL3, helpers.KubectlDelete, helpers.HelperTimeout)
+		ExpectWithOffset(1, err).Should(BeNil(), fmt.Sprintf("Error deleting resource %s: %s", demoPolicyL3, err))
+	}
```
Comment (julianwiedmann) [inline test/k8s/datapath_configuration.go:661]: It's a bit confusing that the patch description speaks about fixing CCNP troubles, but here we also fix a CNP case. Could you split this part off into a second patch? The effective code delta will remain the same, so we can merge without re-running CI. 

Fixing it in the same PR is ok, in the end it's the same pattern of bug.

Comment (jschwinger233) [reply test/k8s/datapath_configuration.go:661]: Will do

Comment (pchaigno) [reply test/k8s/datapath_configuration.go:661]: > so we can merge without re-running CI.

But don't rebase if you want to do that! :smile:

**Review by ldelossa — APPROVED** [review]

**Comment by julianwiedmann** [comment]
No-change push to split the content up into two patches, and CI passed previously.

**Pull request merged**